operator/api/redpanda/v1alpha2/conversion/to_render.go

```diff
@@ -0,0 +1,285 @@
+// Copyright 2025 Redpanda Data, Inc.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.md
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0
+
+package conversion
+
+import (
+	applycorev1 "k8s.io/client-go/applyconfigurations/core/v1"
+	"k8s.io/utils/ptr"
+
+	"github.com/redpanda-data/redpanda-operator/charts/redpanda/v25"
+	"github.com/redpanda-data/redpanda-operator/gotohelm/helmette"
+	redpandav1alpha2 "github.com/redpanda-data/redpanda-operator/operator/api/redpanda/v1alpha2"
+	"github.com/redpanda-data/redpanda-operator/pkg/kube"
+)
+
+// V2Defaults contains the default values for the v2 CRD conversion.
+type V2Defaults struct {
+	RedpandaImage    *redpandav1alpha2.RedpandaImage
+	SidecarImage     *redpandav1alpha2.RedpandaImage
+	ConfiguratorArgs []string
+}
+
+// ConvertV2ToRenderState converts a v2 Redpanda CRD to a redpanda chart RenderState.
+func ConvertV2ToRenderState(config *kube.RESTConfig, defaults *V2Defaults, cluster *redpandav1alpha2.Redpanda, _pools any) (*redpanda.RenderState, error) {
+	spec := defaultV2Spec(defaults, cluster)
+
+	dot, err := redpanda.Chart.Dot(config, helmette.Release{
+		Namespace: cluster.Namespace,
+		Name:      cluster.GetHelmReleaseName(),
+		Service:   "Helm",
+		IsUpgrade: true,
+	}, spec)
+	if err != nil {
+		return nil, err
+	}
+
+	return redpanda.RenderStateFromDot(dot, func(state *redpanda.RenderState) error {
+		return convertV2Fields(state, &state.Values, spec)
+	})
+}
+
+// defaultV2Spec defaults the v2 Redpanda CRD spec to avoid nil dereferences during chart construction.
+func defaultV2Spec(defaults *V2Defaults, cluster *redpandav1alpha2.Redpanda) *redpandav1alpha2.RedpandaClusterSpec {
+	// Big ol' block of defaulting to avoid nil dereferences.
+	spec := cluster.Spec.ClusterSpec.DeepCopy()
+	if spec == nil {
+		spec = &redpandav1alpha2.RedpandaClusterSpec{}
+	}
+
+	if spec.Statefulset == nil {
+		spec.Statefulset = &redpandav1alpha2.Statefulset{}
+	}
+
+	if spec.Statefulset.SideCars == nil {
+		spec.Statefulset.SideCars = &redpandav1alpha2.SideCars{}
+	}
+
+	if spec.Statefulset.SideCars.Controllers == nil {
+		spec.Statefulset.SideCars.Controllers = &redpandav1alpha2.RPControllers{}
+	}
+
+	if spec.Statefulset.InitContainers == nil {
+		spec.Statefulset.InitContainers = &redpandav1alpha2.InitContainers{}
+	}
+
+	if spec.Statefulset.InitContainers.Configurator == nil {
+		spec.Statefulset.InitContainers.Configurator = &redpandav1alpha2.Configurator{}
+	}
+
+	// As we're currently pinned to the v5.10.x chart, we need to set the
+	// default image to the operator's preferred version.
+	spec.Image = defaults.RedpandaImage
```

Comment:
The comment here is no longer correct.

Actually, now that we're using right chart we can just drop this and rely on the "chart's" default once again.